pkg/controller/controller_test.go

```diff
@@ -1852,6 +1855,88 @@ func TestCatalogConversionClusterServicePlanBindable(t *testing.T) {
 
 }
 
+const testCatalogForClusterServiceClassAndPlanWithInvalidExternalIDCharacters = `{
+  "services": [
+    {
+      "name": "mysql",
+      "id": "mysql/1234",
+      "plans": [{
+        "name": "small-mysql",
+        "id": "mysql/100mb"
+      },
+      {
+        "name": "big-mysql",
+        "id": "mysql/100gb"
```

Comment:
add tests that will show conflicts if the escaping isn't done right. For example: mysql/foo & mysql2ffoo for the current PR will show a problem.